### PR TITLE
Optional UTF-8 Decoding (Issue #13)

### PR DESCRIPTION
**Pull request by jimcarreer**

### CODE REVIEW

File: hpack/hpack.py [hunk 1]
```diff
@@ -295,12 +295,16 @@ def header_table_size(self):
...
+    def decode(self, data, raw=False):
         """
         Takes an HPACK-encoded header block and decodes it into a header set.
 
         :param data: A bytestring representing a complete HPACK-encoded header
                      block.
+        :param raw: An optional boolean that when True specifies that raw byte
+                    strings for the name and value should be returned instead
+                    of automatically decoding them as UTF-8 values. The default
+                    value is False.
```
Comment (Lukasa) [inline hpack/hpack.py]: Let's reword this slightly, it's a little tricky to follow.

Something like: "(optional) Whether to return the headers as tuples of raw byte strings or to decode them as UTF-8 before returning them. The default value is False, which returns tuples of Unicode strings."

File: hpack/hpack.py [hunk 2]
```diff
@@ -348,7 +352,10 @@ def decode(self, data):
 
             current_index += consumed
 
-        return [(n.decode('utf-8'), v.decode('utf-8')) for n, v in headers]
+        if raw:
+            return [(n , v) for n, v in headers]
```
Comment (Lukasa) [inline hpack/hpack.py]: Mind removing the space before the comma in the tuple here?

**Comment by Lukasa** [comment]
Fab, a few small notes but this is otherwise perfect. =)

**Comment by jimcarreer** [comment]
Think this should be good to go now.

**Comment by Lukasa** [comment]
\o/ Thanks @jimcarreer! :sparkles: :cake: :sparkles:

**Pull request merged**